.github/workflows/releases.yaml

```diff
@@ -0,0 +1,57 @@
+name: Create Release
+
+on:
+  push:
+    tags:
+      - 'v*.*.*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+      - run: npm install
+      - run: npm build
+      - name: Store build files
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist-files
+          path: |
+            dist
+            src
+            README.md
+            package.json
+            LICENSE
+
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Download build files
+        uses: actions/download-artifact@v2
+        with:
+          name: dist-files
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          files: ./*
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Download build files
+        uses: actions/download-artifact@v2
+        with:
+          name: dist-files
+      - name: Publish on NPM
+        uses: JS-DevTools/npm-publish@v1
+        with:
+          token: ${{ secrets.NPM_TOKEN }}
```

Comment:
@antondalgren is it possible to scope npm tokens so they just have access to a particular project on npm? e.g. the token we use here can only publish amqp-client.js versions

Reply:
Not sure. But there seems to be something called an automation token that we could use. I don't have access to the NPM org and no prior experience with publishing NPM packages.

> Automation: an automation token can download packages and publish new ones, but if you have two-factor authentication (2FA) configured on your account, it will not be enforced. You can use an automation token in continuous integration workflows and other automation systems to publish a package even when you cannot enter a one-time passcode. This is recommended for automation workflows where you are publishing new packages.

https://docs.npmjs.com/creating-and-viewing-access-tokens#creating-access-tokens

Reply:
Great find, that's what we want to use! I'm not sure if we have any npm organization (https://docs.npmjs.com/organizations) or if it is just @carlhoerberg publishing the package with his user

Reply:
It looks like it's published by carls user!

Reply:
i've added a token to secrets, and invited you to the npm org